substrate/runtime/src/lib.rs

```diff
@@ -176,6 +178,20 @@ impl Contains<RuntimeCall> for CallFilter {
       return matches!(call, validator_sets::Call::set_keys { .. });
     }
 
+    if let RuntimeCall::Staking(call) = call {
+      return matches!(
+        call,
+        staking::Call::stake { .. } |
+          staking::Call::unstake { .. } |
+          staking::Call::allocate { .. } |
+          staking::Call::deallocate { .. }
+      );
+    }
+
+    if let RuntimeCall::Session(call) = call {
+      return matches!(call, session::Call::set_keys { .. } | session::Call::purge_keys {});
```

Comment:
umm right. Purge keys takes effect in the next session, so exposing it would remove a validator from the set in the session pallet although we think it is  still in the set in the vs pallet. 👍